xarray/testing.py

```diff
@@ -29,7 +29,8 @@ def wrapper(*args, **kwargs):
         __tracebackhide__ = True
 
         with warnings.catch_warnings():
-            warnings.simplefilter("always")
+            # only remove filters that would "error"
+            warnings.filters = [f for f in warnings.filters if f[0] != "error"]
```

Comment:
can we rely on `warnings.filters`? As far as I can tell, that's semi-public at best: the library is very consistent on using `_` as a prefix for truly private attributes, but it also doesn't document `filters` or list it in `__all__`. However, the standard library doesn't change very often, so I guess it's not too risky...

Reply:
Yes, I am not sure either. But I found no "public" way to unset a _specific_ filter. I think even if it breaks at one point it should mainly affect downstream libraries and not users.

The alternatives are
- always warn
- never warn
- remove the decorator

I would prefer the "result" of what I propose here but I also understand if you deem it too dangerous. I had a quick look at the python issues but found nothing in this direction.

Reply:
`warnings` will only change on upgrade to a new version of `python`, which involves some manual work, anyways (*and* testing is usually blocked by `numba` for a few months).

I guess this means it should be safe to use? @pydata/xarray, what do you think?